docker-java-core/src/main/java/com/github/dockerjava/core/GoLangFileMatch.java

```diff
@@ -52,6 +54,8 @@ private GoLangFileMatch() {
 
     private static final String PATTERN_CHARS_TO_ESCAPE = "\\.[]{}()*+-?^$|";
 
+    private static final Map<String, Pattern> PATTERNS = new HashMap<>();
```

Comment:
Since it is static, I'd suggest using `ConcurrentHashMap` here

Reply:
Good, I changed it to ``java.util.concurrent.ConcurrentHashMap``